future/transaction.go

```diff
@@ -985,4 +1010,43 @@ func parseTxnForeignAssets(foreignAssets []uint64) (parsed []types.AssetIndex) {
 	return
 }
 
+func parseBoxReferences(abrs []types.AppBoxReference, foreignApps []uint64, curAppID uint64) (parsed []types.BoxReference, err error) {
+	for _, abr := range abrs {
+		// there are a few unintuitive details to the parsing:
+		//     1. the AppID of the box must either be in the foreign apps array or
+		//        equal to 0, which references the current app.
+		//     2. if the box references the current app by its appID rather than 0 AND
+		//        the current appID is explicitly provided in the foreign apps array
+		//        then ForeignAppIdx should be set to its index in the array.
+		br := types.BoxReference{Name: abr.Name}
+		found := false
+
+		if abr.AppID == 0 {
+			found = true
+			br.ForeignAppIdx = 0
+		} else {
+			for idx, appID := range foreignApps {
+				if appID == abr.AppID {
+					found = true
+					br.ForeignAppIdx = uint64(idx + 1)
+					break
+				}
+			}
+		}
+
+		if !found && abr.AppID == curAppID {
+			found = true
+			br.ForeignAppIdx = 0
+		}
```

Comment:
nit: want to propose a refactoring: move line 1037 check on `abr.AppID == curAppID` to line 1024, so 2 cases can essentially be merged into 1.

Reply:
my bad, disregard this.